app/models/user.rb

```diff
@@ -1,5 +1,15 @@
 class User < ApplicationRecord
+  enum role: { user: 0, admin: 1 }
   has_secure_password
-  validates_presence_of :firstname, :lastname, :employee_id, :date_of_birth, :joining_day, :designation, :password, message: ->(_, _) { I18n.t('validations.presence') }
+  has_one_attached :profile_photo
+  validates_presence_of :firstname, :lastname, :employee_id, :date_of_birth, :joining_day, :profile_photo, :designation, :role, message: ->(_, _) { I18n.t('validations.presence') }
   validates :email, presence: true, format: { with: /\A[^@\s]+@[^@\s]+\z/, message: ->(_, _) { I18n.t('validations.valid') } }
+
+  def image_resize
+    profile_photo.variant(resize_to_limit: [200, 200]).processed
+  end
+
+  def admin?
+    role == 'admin'
+  end
```

Comment:
I think this method is not needed. Rails gives you a way to find enums efficiently like, user.admin? or user.user?

Check this: https://api.rubyonrails.org/v5.1/classes/ActiveRecord/Enum.html